public/components/layer_control_panel/hide_layer_button.tsx

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiButtonIcon, EuiFlexItem } from '@elastic/eui';
+import React, { useState } from 'react';
+import { Map as Maplibre } from 'maplibre-gl';
+import {
+  LAYER_PANEL_HIDE_LAYER_ICON,
+  LAYER_PANEL_SHOW_LAYER_ICON,
+  LAYER_VISIBILITY,
+} from '../../../common';
+import { MapLayerSpecification } from '../../model/mapLayerType';
+import { updateLayerVisibility } from '../../model/map/layer_operations';
+
+interface MaplibreRef {
+  current: Maplibre | null;
+}
+
+interface HideLayerProps {
+  layer: MapLayerSpecification;
+  maplibreRef: MaplibreRef;
+}
+export const HideLayer = ({ layer, maplibreRef }: HideLayerProps) => {
+  const [layerVisibility, setLayerVisibility] = useState(new Map<string, boolean>([]));
+
+  const onLayerVisibilityChange = (selectedLayer: MapLayerSpecification) => {
+    if (selectedLayer.visibility === LAYER_VISIBILITY.VISIBLE) {
+      selectedLayer.visibility = LAYER_VISIBILITY.NONE;
+      setLayerVisibility(new Map(layerVisibility.set(layer.id, false)));
+    } else {
+      layer.visibility = LAYER_VISIBILITY.VISIBLE;
+      setLayerVisibility(new Map(layerVisibility.set(layer.id, true)));
+    }
+    updateLayerVisibility(maplibreRef.current!, layer.id, layer.visibility);
```

Comment:
We shouldn't be modifying props directly. Treat props as read only and pass mutation as function from parent.

Reply:
Updated to use updateLayer(updatedLayer) to change the property from parent